src/controllers/LoginController.php

```diff
@@ -303,32 +308,47 @@ private function getAuthService(string $authType): AuthInterface
                     ),
                     $this->App->Request->request->getAlnum('mfa_code'),
                 );
+            case 'teaminit':
+                $this->initTeamSelection();
+                exit;
+            case 'teamselection':
+                $this->teamSelection($this->App->Session->get('teaminit_userid'), $this->App->Request->request->getInt('team_id'));
+                exit;
 
             default:
                 throw new ImproperActionException('Could not determine which authentication service to use.');
         }
     }
 
-    private function initTeamSelection(string $authType): void
+    /**
+     * For when a user already exists but has no associated team
+     */
+    private function teamSelection(int $userid, int $teamId): void
     {
-        if ($authType === 'teaminit'
-            && $this->App->Session->get('initial_team_selection_required')
-        ) {
-            // create a user in the requested team
-            $newUser = ExistingUser::fromScratch(
-                $this->App->Session->get('teaminit_email'),
-                array($this->App->Request->request->getInt('team_id')),
-                $this->App->Request->request->getString('teaminit_firstname'),
-                $this->App->Request->request->getString('teaminit_lastname'),
-            );
-            $this->App->Session->set('teaminit_done', true);
-            // will display the appropriate message to user
-            $this->App->Session->set('teaminit_done_need_validation', (string) $newUser->needValidation);
-            $this->App->Session->remove('initial_team_selection_required');
-            $location = '/login.php';
-            echo "<html><head><meta http-equiv='refresh' content='1;url=$location' /><title>You are being redirected...</title></head><body>You are being redirected...</body></html>";
-            exit;
-        }
+        $this->App->Session->remove('team_selection_required');
```

Comment:
Are there other session settings that can be removed here, eg `teaminit_userid` (which seems to be passed as the userid to this function)?

In particular, once you've selected a team, you should probably not be able to add yourself to more teams by manually poking the LoginController.

This is not based on dynamic testing, only skimming the pull request.